tests/functional/m/modified_iterating.py

```diff
@@ -4,35 +4,35 @@
 import copy
 from enum import Enum
 
-item_list = [1, 2, 3]
-for item in item_list:
-    item_list.append(item)  # [modified-iterating-list]
+ITEM_LIST = [1, 2, 3]
+for item in ITEM_LIST:
+    ITEM_LIST.append(item)  # [modified-iterating-list]
 
-for item in item_list:
-    item_list.remove(item)  # [modified-iterating-list]
+for item in ITEM_LIST:
+    ITEM_LIST.remove(item)  # [modified-iterating-list]
 
-for item in item_list.copy():
-    item_list.append(item)
-    item_list.remove(item)
+for item in ITEM_LIST.copy():
+    ITEM_LIST.append(item)
+    ITEM_LIST.remove(item)
 
-for item in copy(item_list):
-    item_list.append(item)
-    item_list.remove(item)
+for item in copy(ITEM_LIST):
+    ITEM_LIST.append(item)
+    ITEM_LIST.remove(item)
 
-for item in [k for k in item_list]:
-    item_list.append(item)
-    item_list.remove(item)
+for item in [k for k in ITEM_LIST]:
+    ITEM_LIST.append(item)
+    ITEM_LIST.remove(item)
```

Comment:
We're modifying a supposed constant here ? Look like a false negative ?

Reply:
It's still a constant even if it's mutable, right?

Reply:
I think constants should be immutable but that might be an issue with the definition of what a constant is supposed to be in python (Time to read some PEP I guess?). I think I remember some argument by a user that this message will always have false positives / false negatives because it's "impossible" to know if a python object is an immutable or not. I kinda like the approach of determining if something is a constant if it's reassigned or muted in the actual code regardless of the actual immutable/mutable property of the instance we're analyzing.

Reply:
It's a judgment call for sure, but it strikes me as a little fancy to do that much checking. A list that gets appended to a couple times depending on a couple of if conditions at the top of a module is still a constant when another module imports it.

I'm open to other views, so no rush here, but I don't foresee this being much of a problem in practice (other than needing to silence invalid name a couple times in a Django settings.py file assuming it's even linted -- e.g. for:

if DEBUG:
    SILENCED_SYSTEM_CHECKS += [...]

Reply:
Hmm, now that you say it, I definitely mutate constants in django setting's files. It think I remember django rosetta suggesting this at some point:

```python
if DEBUG:
    INSTALLED_APPS.append("rosetta")
```
It would be annoying to silence this all the time.  (Maybe something for pylint-django to take care off if we choose to go this way.)  Let's wait for other opinions to do it. But in any case this MR is already a huge improvement, we could merge as is (and create a discussion issue to be resolved before we release 4.0?).

Reply:
Actually, that wouldn't even be a problem, because `append()` doesn't reassign. It mutates, but that's irrelevant for this check. I used `+=` just to focus on reassignment.

Reply:
/cc @zenlyj

Reply:
I was arguing that constants shouldn't be mutated either, if we do the check so that "mutated variable are not constant so they should be snake case" then it would start to be a problem in django settings too.

Reply:
> if we do the check so that "mutated variable are not constant so they should be snake case"

Maybe I'm missing what you're saying - I would be against such a check, since I think a mutated constant is still a constant.

Reply:
Yes we disagree on this. [PEP8 is very vague about this, there's no definition of what a constant is](https://peps.python.org/pep-0008/#constants). After searching for ~= 15mn I didn't find anything authoritative to support or refute my opinion in the PEP, so I welcome opinion from anyone reading this.

Imo a constant value should not change during the execution of the program. Mutating means it's modified during execution so it contradicts the idea of being a constant. (In the case of django the settings.py is read/executed once for each server launch so it's not wrong to call it a constant _after_ the settings.py is executed and the server is launched).

Reply:
Personally I would be cautious about mutating constants in runtime like what @Pierre-Sassoulas has mentioned, but there seems to be actual use cases as mentioned above (in Django settings).

Since there is no official definition in PEP, I think it all boils down to how strictly we want to treat constants, it could go both ways really. @jacobtylerwalls is probably thinking about constants like how the `const` keyword works in Javascript? Where a constant declaration cannot be reassigned but can be mutated. At the same time, we have languages like CPP that take it one step further by enforcing immutability on constant objects.

Reply:
Yeah, the `const` keyword in javascript is exactly how I'm thinking about it. Plus, python explicitly has [no concept of immutable objects](https://docs.python.org/3/library/dataclasses.html#frozen-instances), so "constant" is really more of a shorthand. Maybe it's unfortunate we're discussing the word "constant" at all. That's why I'm preferring to speak about "reassignments".